### PR TITLE
Compacted topics feature

### PR DESCRIPTION
**Pull request by ivanwakeup**

### CODE REVIEW

File: kafka/src/main/resources/reference.conf [hunk 1]
```diff
@@ -93,6 +95,14 @@ hydra {
       key.deserializer = org.apache.kafka.common.serialization.StringDeserializer
       value.deserializer = io.confluent.kafka.serializers.KafkaAvroDeserializer
     }
+
+    //compacted streams config
+    parallelism = 3
+    eos-commit-interval = 100
```
Comment (ivanwakeup) [inline kafka/src/main/resources/reference.conf]: we aren't using a transactional sink, but eos-commit-interval is needed by CommiterSettings

**Comment by roarking** [comment]
Lies.

File: kafka/src/main/resources/reference.conf (hunk 1 above)
Comment (znickle24) [reply kafka/src/main/resources/reference.conf]: look at the parallelism to verify it's correct

File: kafka/src/main/scala/hydra/kafka/services/StreamsManagerActor.scala [hunk 2]
```diff
@@ -52,25 +58,49 @@ class MetadataConsumerActor(consumerConfig: Config,
...
     case StopStream =>
       pipe(stream._1.shutdown().map(_ => StreamStopped)) to sender
+
+    case GetStreamActor(actorName: String) =>
+      val s = sender()
+      pipe(Future{GetStreamActorResponse(context.child(actorName))}) to s
   }
+
+
+  private[kafka] def maybeCreateCompactedStream(metadata: TopicMetadata): Unit = {
```
Comment (znickle24) [inline kafka/src/main/scala/hydra/kafka/services/StreamsManagerActor.scala]: Make this return Props

**Comment by ivanwakeup** [comment]
REQUESTED CHANGES:
1. add back wrapper types for return values in TopicBootstrapActor (Eg. BootstrapSuccess, BootstrapFailure)
2. refactor maybeCreateCompactedStream to return Props of compactedStreamActor, then context.actorOf in receive method
3. add explicit shutdown behavior for compactedStreamActor
4. verify Commit settings for compactedTopicStream (at least once delivery)

**Comment by ivanwakeup** [comment]
Akka docs on at-least-once semantics for committing topic-to-topic streams.

https://doc.akka.io/docs/akka-stream-kafka/current/consumer.html#connecting-producer-and-consumer

@roarking

**Pull request merged**